src/main/java/com/edu/Hint.java

```diff
@@ -0,0 +1,45 @@
+package com.edu;
+
+public class Hint {
+
+    public String hintInfo(String myNum, String comNum) {
+
+        int strike = 0;
+        int ball = 0;
+        int validationCnt = 0;
+        int myNumLength = myNum.length();
+        String hintAnswer = "";
+        MyNumber myNumber = new MyNumber();
+
+        validationCnt = myNumber.validation(myNum,validationCnt);
+
+        if (validationCnt < 1) {
+
+            for (int i = 0; i < myNumLength; i++) {
+                if (comNum.charAt(i) == myNum.charAt(i)) {
+                    strike++;
+                }
+            }
+
+            for (int i = 0; i < myNumLength; i++) {
+                if (comNum.contains(myNum.substring(i, i + 1))) {
+                    ball++;
+                }
+            }
+
+            if (ball == 0) {
+                hintAnswer = "낫싱";
+            } else if (strike > 0 && strike - ball == 0) {
+                hintAnswer = strike + " 스트라이크";
+            } else if (strike > 0 && ball > 0 && ball - strike > 0) {
+                hintAnswer = strike + " 스트라이크 " + (ball - strike) + "볼";
+            } else {
+                hintAnswer = ball + "볼";
+            }
+
+        }
+
+        return hintAnswer;
```

Comment:
이 코드는 아래와 같이 리팩토링 할 수 있습니다. 
어떤 점이 바뀌었는지 그리고 이로인한 장점은 무엇일지 고민해보고 댓글로 남겨두면 도움이 될 것 같습니다. 

```suggestion
        if (validationCnt >= 1) {
            return "";
        }

        int strike = 0;
        int myNumLength = myNum.length();
        for (int i = 0; i < myNumLength; i++) {
            if (comNum.charAt(i) == myNum.charAt(i)) {
                strike++;
            }
        }

        int ball = 0;
        for (int i = 0; i < myNumLength; i++) {
            if (comNum.contains(myNum.substring(i, i + 1))) {
                ball++;
            }
        }

        if (ball == 0) {
            return "낫싱";
        }

        if (strike > 0 && strike - ball == 0) {
            return strike + " 스트라이크";
        }

        if (strike > 0 && ball > 0 && ball - strike > 0) {
            return strike + " 스트라이크 " + (ball - strike) + "볼";
        }

        return ball + "볼";
    }
```

여기서 추가적으로 **ball을 카운트하는 기능**과, **strike를 count 하는 기능**을 메서드로 추출한다면 hintInfo()메소드의 코드 수를 더 줄일 수 있을 것입니다.